### DISCUSSION TITLE
Multibooking form for family members and bulk change of seats

### DISCUSSION THREAD

**Discussion by refugeindhamma** [liedekef/events-made-easy#85] [Q&A] [answered]
Hi Franky,

I'm currently creating a Multibooking form for Kids' course spanning a few months but have faced 2 difficulties:

1) Parents' + children's info
In the Multibooking form, I would like to have one of the parent's particulars, then for the parent to indicate how many children they want to enroll for the course. On top of that, I also need the children's names. 

2) Bulk change of seats
Is it possible to bulk change seats for the entire course of multiple lessons, i.e. only 1 selector for number of seats and it is automatically applied to all lessons? As the course lasts a few months, it will be very tedious for parents to select seats for every single lesson.

Point 2 is actually linked to point 1. I was thinking if it's possible to have a single seat selector for all the lessons in the Multibooking form, then in addition, have a name text box appear for 1 seat, 2 name text boxes for 2 seats, etc. This is for parents to enter the names of their children attending the course.

Alternatively, if I don't use Multibooking form for the course and use the single booking form for the whole duration of the course, can the QR code generated be scanned multiple times during the duration of the course? For example the course having 10 lessons, can the QR code be scanned 10 times for every seat registered?

**Comment by liedekef** [comment, marked as answer]
Currently EME doesn't do family info in RSVP forms, only in memberships (feel free to sponsor :-) )
However: you can ask for the child's name in an extra custom field (of type 'text' I'd say).
Next to that, there's no bulk change of seats ==> but anybody with some jquery can cook that up themselves. Also: you have the "simple" multibooking form as well, and/or make it an attendance form (min seats to book=0, max seats to book=1, attendance=yes). So much to try :-)
Concerning the single booking form: the QR code can be scanned the same amount of times as there are seats booked.

**Reply by refugeindhamma** [reply]
Thanks. I will give your suggestions a try.

**Reply by refugeindhamma** [reply]
Just to leave a record here on the solution I am using based on Franky's suggestions.

I am using min seats = 1, max seats =1 and attendance = Yes. This is so that registrants do not have to check the checkboxes that appear with min seats = 0, max seats = 1 and attendance = Yes.

I have also created a custom "Textarea" field for parents to input their children's names so that parents can enter more than 1 child's name.

**Reply by refugeindhamma** [reply]
However, I have one more question: 

Now, I am using only #_{d/m} #_SEATS in the template as this is shortest info that makes sense for the registrant. My current course will be held from June to Oct 2023 but for next year, it will run weekly for the whole year for around 10 months. The dates will be extremely long in the Multibooking form and confirmation emails. I have even used CSS to organise the list into 4 columns but with at least 10 x 4 = 40 dates, it is still going to look very long and disorganised:
7 Jan, 14 Jan, 28 Jan, 4 Feb,
11 Feb, 18 Feb, 25 Feb, 4 Mar,
11 Mar, 18 Mar, 25 Mar, 1 Apr, etc

So, is it possible to shorten the date in the Multibooking form Body template? By shortening it, I mean like grouping the dates together in months (example below). With this grouping of dates into monthly categories, it will look more organised so it will be like:
Jan 7, 14, 28
Feb 4, 11, 18, 25
Mar 4, 11, 18, 25
Apr 1, etc.

Can the same also be done for confirmation/confirmed emails? Instead of listing every single seat booked, just have 1 sentence saying something like:
You are booking/have booked 1 seat for #_EVENT on the following dates:
Jan 7, 14, 28
Feb 4, 11, 18, 25
Mar 4, 11, 18, 25
Apr 1, etc.

**Reply by liedekef** [reply]
sorry, but I have no real solution to visualisation prob, unless you take a look at the available filters and code it up yourself. Also: attendance is not really needed if min and max=1 (since then it is hidden anyway), but can't hurt.
